lib/blob/blobstore.c

```diff
@@ -8149,11 +8163,16 @@ bs_snapshot_checksum_blob_read_cpl(void *cb_arg, int bserrno)
 		SPDK_ERRLOG("blob 0x%" PRIx64 " snapshot checksum, blob read error %d\n", ctx->blob->id, bserrno);
 		ctx->bserrno = bserrno;
 		_blob->locked_operation_in_progress = false;
+		bs_snapshot_free_clusters_checksums(_blob);
 		spdk_blob_close(_blob, bs_snapshot_checksum_cleanup_finish, ctx);
 		return;
 	}
 
 	ctx->checksum = spdk_crc64_iso_refl(ctx->read_buff, _blob->bs->cluster_sz, ctx->checksum);
+	if (_blob->clusters_checksums) {
+		_blob->clusters_checksums[ctx->cluster] = spdk_crc64_iso_refl(ctx->read_buff, _blob->bs->cluster_sz,
+				0);
+	}
```

Comment:
Does this mean that spdk_tgt must calculate twice after each one-time read? 

I am thinking, if we can use the existing `_blob->clusters_checksums[ctx->cluster]` and `ctx->checksum` to quickly update `ctx->checksum`

Reply:
Atually it is not possible, inside `isa-l`, which is the library SPDK uses to calculate the checksum, there isn't any function to do so.
But the suite [crcany](https://github.com/madler/crcany) is able to generate a combining function that we could use to avoid the double calculation, it can be a future improvement.

Reply:
> But the suite [crcany](https://github.com/madler/crcany) is able to generate a combining function that we could use to avoid the double calculation, it can be a future improvement.

Great! Would you create a GitHub ticket for this improvement?

Reply:
Done: https://github.com/longhorn/longhorn/issues/10871